test/sequential/sequencerStaking.rewards.test.ts

```diff
@@ -0,0 +1,299 @@
+    ethUser2.keyRingPair.address,
+  );
+  const user3AwardedPts = await SequencerStaking.awardedPts(
+    rewardsSessionNumber,
+    testUser3.keyRingPair.address,
+  );
+  expect(user1AwardedPts).bnEqual(pointsValue.divn(3));
+  expect(user2AwardedPts).bnEqual(BN_ZERO);
+  expect(user3AwardedPts).bnEqual(pointsValue.divn(3).muln(2));
+  let sessionIndex = await getSessionIndex();
+  while (sessionIndex <= rewardsSessionNumber + 1) {
+    await waitNewStakingRound();
+    sessionIndex = await getSessionIndex();
+  }
+  await ethUser1.refreshAmounts(AssetWallet.BEFORE);
+  await ethUser2.refreshAmounts(AssetWallet.BEFORE);
+  await testUser3.refreshAmounts(AssetWallet.BEFORE);
+  const payoutEvent1 = await Sudo.asSudoFinalized(
+    Sudo.sudoAs(
+      ethUser1,
+      SequencerStaking.payoutRewards(ethUser1.keyRingPair.address, 2),
+    ),
+  );
+  expectMGAExtrinsicSuDidSuccess(payoutEvent1);
+  const filteredEvent1 = await filterAndStringifyFirstEvent(
+    payoutEvent1,
+    "Rewarded",
+  );
+  const sequencerRewards1 = stringToBN(filteredEvent1[2]);
+  const payoutEvent2 = await Sudo.asSudoFinalized(
+    Sudo.sudoAs(
+      ethUser2,
+      SequencerStaking.payoutRewards(ethUser2.keyRingPair.address, 2),
+    ),
+  );
+  expectMGAExtrinsicSuDidFailed(payoutEvent2);
+  const payoutEvent3 = await Sudo.asSudoFinalized(
+    Sudo.sudoAs(
+      testUser3,
+      SequencerStaking.payoutRewards(testUser3.keyRingPair.address, 2),
+    ),
+  );
+  expectMGAExtrinsicSuDidSuccess(payoutEvent3);
+  const filteredEvent3 = await filterAndStringifyFirstEvent(
+    payoutEvent3,
+    "Rewarded",
+  );
+  const sequencerRewards3 = stringToBN(filteredEvent3[2]);
+  await ethUser1.refreshAmounts(AssetWallet.AFTER);
+  await ethUser2.refreshAmounts(AssetWallet.AFTER);
+  await testUser3.refreshAmounts(AssetWallet.AFTER);
+  const diff1 = ethUser1.getWalletDifferences()[0].diff.free;
+  const diff2 = ethUser2.getWalletDifferences();
+  const diff3 = testUser3.getWalletDifferences()[0].diff.free;
+  expect(diff3).bnGt(diff1);
+  expect(diff2).toBeEmpty();
+  expect(diff1).bnEqual(sequencerRewards1);
+  expect(diff3).bnEqual(sequencerRewards3);
+  expect(sequencerRewards1).bnEqual(
+    sequencerRewards3.divn(2) || sequencerRewards3.divn(2).addn(1),
+  );
+});
+
+it("Regardless joining , slash, join or leaving sequencer set, Sequencer will be paid if points", async () => {
+  await Sudo.batchAsSudoFinalized(Assets.FinalizeTge(), Assets.initIssuance());
+  await createAnUpdate(testUser, chainEth);
+  const rewardsSessionNumber = await getSessionIndex();
+  await leaveSequencing(testUser.keyRingPair.address);
+  const sequencersList = await SequencerStaking.activeSequencers();
+  expect(sequencersList.toHuman().Ethereum).not.toContain(
+    testUser.keyRingPair.address,
+  );
+  let sessionIndex = await getSessionIndex();
+  while (sessionIndex <= rewardsSessionNumber + 1) {
+    await waitNewStakingRound();
+    sessionIndex = await getSessionIndex();
+  }
```

Comment:
When we divide the reward into three parts it may turn out that the reward is equal, then when dividing into 3 parts and 2 users we will get numbers that differ by one. I am not completely sure how this works in the system, but I received a reward for one and another sequencer, say, 630001 and 315000, while I wanted to leave the most correct check that these are not random numbers, namely 2/3 and 1/3 of the total reward

Reply:
Done